### PR TITLE
Add qdevice

### PR DESCRIPTION
**Pull request by aleksei-burlakov**
@arbulu89 @MalloZup @Simranpal Please have a look. This PR should come together with https://github.com/SUSE/salt-shaptools/pull/74

### CODE REVIEW

**Review by MalloZup — DISMISSED** [review]
hey @aleksei-burlakov  thx for pr! looks good

I think we need to add 2 things:

- `pillar.example`:  we need to extend that file with the new var https://github.com/SUSE/habootstrap-formula/blob/master/pillar.example

e.g https://github.com/SUSE/habootstrap-formula/blob/master/pillar.example#L21 in this direction

- documentation : I would add here a line https://github.com/SUSE/habootstrap-formula#features

File: cluster/packages.sls
Comment (MalloZup) [inline cluster/packages.sls]: Regarding FIXME , my personal view on this:

generally I'm don't have a strong opinion on introducing them or not using them.

But if we use them we should refer to a link or issue which track the upstream blocker.


like  in our case something 

`# FIXME! This should be in the patters-ha-ha_sles #jsc-139123` which refers to JIRA whatever

File: cluster/packages.sls
Comment (arbulu89) [inline cluster/packages.sls]: Here I would make this installation conditional. If `qdevice` data is present in the pillar file, we install it, not otherwise

**Review by arbulu89 — COMMENTED** [review]
@aleksei-burlakov Is there any reason why `qdevice` is a map instead of just the `hostname`.
Is this because we can configure qdevice in different ways?
Just to understand the need of this nested information

**Comment by aleksei-burlakov** [comment]
> @aleksei-burlakov Is there any reason why `qdevice` is a map instead of just the `hostname`.
> Is this because we can configure qdevice in different ways?
> Just to understand the need of this nested information

Of course, it also supports other parameters, like port, tie-breaker, and a tls flag.

File: cluster/packages.sls
Comment (aleksei-burlakov) [reply cluster/packages.sls]: I've made installing the `corosync-qdevice` conditional.

File: cluster/packages.sls
Comment (aleksei-burlakov) [reply cluster/packages.sls]: Good idea, I made it installing the `corosync-qdevice` conditional.

**Comment by aleksei-burlakov** [comment]
@arbulu89, @MalloZup, @Simranpal please check out the changes?

**Review by arbulu89 — APPROVED** [review]
Thanks @aleksei-burlakov 
Let's wait until having SUSE/salt-shaptools#74 ready to merge both together

File: cluster/packages.sls
Comment (smpsgill) [inline cluster/packages.sls:43]: These conditions might be combined, but I see in future if there could be more parameters under `cluster.qdevice` So this is fine

**Review by smpsgill — APPROVED** [review]
Looks good @aleksei-burlakov

**Comment by arbulu89** [comment]
@aleksei-burlakov Add what this PR brings in the changelog please (the changes file). You don't need to increase the version, just write what it does.

File: cluster/packages.sls
Comment (aleksei-burlakov) [reply cluster/packages.sls:43]: Thanks, yes, the qdevice has more arguments that should come in future commits.

**Review by arbulu89 — APPROVED** [review]
@aleksei-burlakov Add the new functionality description in the changes file (don't bump the version, add just a text saying what it does).
After that, I think we are ready to merge

**Pull request merged**
